### ISSUE TITLE
APT Dependencies Incomplete for Ubuntu 14.04

### ISSUE THREAD

**Issue by GroundPound** [BinaryAnalysisPlatform/bap#78]
I started with the official [Ubuntu Trusty 64](https://atlas.hashicorp.com/ubuntu/boxes/trusty64) Vagrant box and followed the instructions in the [BAP README](https://github.com/BinaryAnalysisPlatform/bap/blob/master/README.md). I found that the list of [system package dependencies](https://github.com/BinaryAnalysisPlatform/bap/blob/master/apt.deps) was incomplete.

Installing OPAM from the Ubuntu package manager did not install `ocaml-native-compilers`. The BAP installation failed while building the `ocamlfind` dependency and I had to revert OPAM and begin the installation again. Additionally I needed the `m4` package for a different BAP dependency. Unfortunately I didn't write down which one in my notes, but I had to revert OPAM and begin the BAP installation again.

The last error I had before I gave up:

```
===== ERROR while installing ssl.0.4.7 =====
# opam-version 1.1.1
# os           linux
# command      ./configure --prefix /home/vagrant/.opam/system
# path         /home/vagrant/.opam/system/build/ssl.0.4.7
# compiler     system (4.01.0)
# exit-code    1
# env-file     /home/vagrant/.opam/system/build/ssl.0.4.7/ssl-21466-427793.env
# stdout-file  /home/vagrant/.opam/system/build/ssl.0.4.7/ssl-21466-427793.out
# stderr-file  /home/vagrant/.opam/system/build/ssl.0.4.7/ssl-21466-427793.err
### stdout ###
# ...[truncated]
# checking for ocamldep... /usr/bin/ocamldep
# checking for ocamllex... /usr/bin/ocamllex
# checking for ocamlyacc... /usr/bin/ocamlyacc
# checking for ocamldoc... /usr/bin/ocamldoc
# checking for ocamlmktop... /usr/bin/ocamlmktop
# checking for gcc... (cached) gcc
# checking whether we are using the GNU C compiler... (cached) yes
# checking whether gcc accepts -g... (cached) yes
# checking for gcc option to accept ISO C89... (cached) none needed
# checking for SSL_new in -lssl... no
### stderr ###
```

One last note that may be worth mentioning is to recommend users in the README run `opam update` before the installation. The [installation instructions](https://opam.ocaml.org/doc/Install.html) don't say to do that, and it took me a while to figure out why BAP was an "invalid package name." The corresponding `apt-get` message is `Unable to locate package`.

**Comment by rvantonder** [comment]
+1 for `opam update`, this bit me before as well.

**Comment by ivg** [comment]
The only missed dependency is `clang++` compiler, that is used to compile bap on different platforms. What concerning other dependencies, they are not from bap. And if I will follow the transitive closure of dependencies, then I need also to include instructions on how to install linux, or mac os x. 

Now seriously, the instructions presumes that you have a working opam installation. For some reason, you managed to broke opam, or, forget to activate it.

**Comment by dbrumley** [comment]
It sounds from Peter like two people were not able to install because of this, and given that opam is new, I think we should add to the readme that you should run opam update to be up2date.

**Comment by ivg** [comment]
fixed in `0.9.2` release.

although I've forgotten to write about `opam update` in the readme file.
